src/libs/actions/BankAccounts.js

```diff
@@ -602,9 +618,18 @@ function setupWithdrawalAccount(data) {
     }
 
     API.BankAccount_SetupWithdrawal(newACHData)
+        /* eslint-disable arrow-body-style */
```

Comment:
This is weird, I think it should maybe be like this instead?

```javascript
// First, optimistically update the reimbursement account in Onyx
Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {
    loading: false,
    achData: {...newACHData},
})

    // Then update it in the API
    .then(() => API.BankAccount_SetupWithdrawal(newACHData))
    .then((response) => {
        // Do other stuff
    });
```

Reply:
Also probably want to add a `catch` block to undo the optimistic change if the API call doesn't work for some reason.

Reply:
Or also undo the optimistic change if the API returns an error response.

Reply:
I agree that's the ideal flow, but I'm a bit hesitant to make that change right now since it will affect every step in the VBA flow. We currently have https://github.com/Expensify/Expensify/issues/173363 to ensure we go back and refactor the flow to make it easier to understand and less buggy, perhaps we can make this improvement there?

Reply:
> Also probably want to add a catch block to undo the optimistic change if the API call doesn't work for some reason.

This is actually already done [here](https://github.com/Expensify/App/blob/f5bb5345f0319deeec2d9d64d78363a3daecadcf/src/libs/actions/BankAccounts.js#L756), so we should be good there.

> Or also undo the optimistic change if the API returns an error response.

We have some minimal error handling set up, but adjusting this can be out-of-scope for this PR. Even in its current form, this PR does does this before checking the API response status code:

```javascript
Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {
    loading: false,
    achData: {...newACHData},
})
```

So doing that before the API call or after should make no difference to the error handling.

All this is to say that the only change that needs to be made is the one from my first comment in this thread, which imo is pretty simple and safe to make. The only thing happening between the time when I'm suggesting we do the `Onyx.set` and when you're currently doing the `Onyx.set` is an API call, and the API call has no dependence on the Onyx data, so we should be good. We just have our promise chain out-of-order in this PR in its current form.

Reply:
> We have some minimal error handling set up, but adjusting this can be out-of-scope for this PR.

Sounds good, I'll have a look at updating just the API call order and doing some extra testing to make sure everything still works.

Reply:
Did some testing here and making that update will result in the UI being blocked on the API call since the UI updates all happen in the `.then` handler after the API call is done, so the loading spinner isn't shown and the screen looks frozen. To update this, we'd have to do everything optimistically outside of the then function, and then revert the data when the API call fails. Alternatively, we can remove the `loading: false` from the merge call and add it in a separate merge call after the API call comes back, but that'll be pretty much the same thing we're doing now, so I don't think this is worth updating without a full refactor.

Reply:
Okay, fair enough. Let's make a note of this in https://github.com/Expensify/Expensify/issues/173363